### PR TITLE
GEN-1459 | Pre-fill customer data from search params

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

- Pick up customer data from URL and update shop session

- Pass along remaining search params to the next route

## Justify why they are needed

- Support for the partner to pass along customer data in the URL when starting the flow

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #3564** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3564?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #3565** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3565?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3565?utm_source=stack-comment).

File: apps/store/src/features/widget/parseSearchParams.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  let ssn: string | undefined = undefined
+  if (searchSsn && Personnummer.valid(searchSsn)) {
+    ssn = Personnummer.parse(searchSsn).format(true)
+    updatedSearchParams.delete(SearchParam.Ssn)
+  }
+
+  const email = searchParams.get(SearchParam.Email)
+  if (email) updatedSearchParams.delete(SearchParam.Email)
+
+  if (!ssn && !email) return [null, updatedSearchParams]
```
Comment (alebedev) [inline apps/store/src/features/widget/parseSearchParams.ts:27]: Nit: maybe returning empty object is a cleaner way to represent it?

File: apps/store/src/pages/widget/flows/[...slug].tsx [hunk 2]
```diff
@@ -56,17 +60,38 @@ export const getServerSideProps: GetServerSideProps<any, Params> = async (contex
     variables: { input: { countryCode, partnerName: story.content.partner } },
   })
 
-  const shopSessionId = result.data?.shopSessionCreatePartner.id
-  if (!shopSessionId) throw new Error('Missing shopSessionId')
+  if (!result.data) throw new Error('Failed to create Partner Shop Session')
+
+  const [customerData, updatedSearchParams] = parseCustomerDataSearchParams(url.searchParams)
+  if (customerData) {
```
Comment (alebedev) [inline apps/store/src/pages/widget/flows/[...slug].tsx:66]: Should this parse be extracted into its own method?

**Review by alebedev — APPROVED** [review]
Great work!  Some styling suggestions inside

File: apps/store/src/features/widget/parseSearchParams.ts (hunk 1 above)
Comment (robinandeer) [reply apps/store/src/features/widget/parseSearchParams.ts:27]: Haha yeah that's how I first did it actually -- okey let's try that instead

Comment (alebedev) [reply apps/store/src/features/widget/parseSearchParams.ts:27]: You decide :)  If variant with null looks better, we can use it

Maybe just replacing `if (!ssn && !email)` with empty object check should be enough.  What scares me a little about current implementation is that there're two separate places to keep track of field list we're working with

**Comment by robinandeer** [comment]
### Merge activity

* **Nov 15, 10:16 AM**: [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3565) rebased this pull request after merging its parent, because this pull request is set to merge when ready.

**Pull request merged**